addons/escoria-core/game/core-scripts/esc/esc_object_manager.gd

```diff
@@ -16,95 +16,297 @@ const RESERVED_OBJECTS = [
 ]
 
 
-# The hash of registered objects (the global id is the key)
-var objects: Dictionary = {}
+# The array of registered objects (organized by room, so each entry is a structure
```

Comment:
I was asked by @balloonpopper to elaborate and clarify the structure. I'd suggest that having more detail is warranted. If you still disagree, I'll look at changing this.

Reply:
I just dislike such HUGE docs in code. It makes the code hard to read. I'd be okay if we just remove the example. Would you be okay with that as well, @balloonpopper?

Or we could add a documentation page about the object manager to the docs.

Reply:
I'm happy if we change this to a couple of sentences and shift the body of this elsewhere (as long as there's a "see (link)[...]" for more details to make it easy to find). I would however keep the example if you're keeping the code. (1) it brings new developers quickly up to speed, and (2) it brings old developers back up to speed when they don't look at this for a while and can't remember how it works :-) 
I'm a fan of having example structures as part of the code - why have to reverse engineer what something will look like if you can say "it should look like this" - debugging is then easy as you can quickly determine whether the data structure looks like it should.
I think the benefits of large descriptions far outweigh the downsides (as long as they're not mid-function). I also fall in the "would like to see almost every line of code with a comment" category though.

Reply:
I think similarly to @balloonpopper, at least insofar as the example and descriptions go.

That said, I'm willing to entertain suggestions on how to improve this and make everyone happy.

Reply:
My main concern here is that the json representation of the structure is misleading because it's actually a list of Godot objects and that it's far too big so as a new developer looking at the code you're greeted with this *huge* wall of text.

So, yeah, would be pretty much voting for the "move it to the docs" solution.

Reply:
It's more a generic object representation than proper JSON, but I think I understand what you're getting at.

@dploeger can I suggest creating a separate issue for this so we can get this PR merged?

Reply:
Yes, fine with me!